msal-dotnet-articles/advanced/monitoring.md

```diff
@@ -5,44 +5,151 @@ description: "How to use the metrics provided by MSAL.NET to monitor your applic
+One of the tools MSAL provides to combat production issues is logging errors when MSAL in not properly configured. It is critical to enable logging whenever possible to not only monitor logs for errors but also help in the diagnosis of problematic events. See [Logging in MSAL.NET](./exceptions/msal-logging.md) for details.
 
 The following errors will be logged in MSAL:
 
-- When using an authority ending in `/common` or `/organizations` for client credential authentication. `AcquireTokenForClient` (This will be available in 4.37.0)
+- When using an authority ending in `/common` or `/organizations` for client credential authentication (<xref:Microsoft.Identity.Client.ConfidentialClientApplication.AcquireTokenForClient(System.Collections.Generic.IEnumerable{System.String})>).
   - The current authority is targeting the `/common` or `/organizations` endpoint which is not recommended. See [Client credential flows](../acquiring-tokens/web-apps-apis/client-credential-flows.md) for more details.
-- When the default internal token cache is used when using confidential client applications.
-  - The default token cache provided by MSAL is not designed to be performant when used in confidential client applications. Please use token cache serialization. See [Token cache serialization in MSAL.NET](/azure/active-directory/develop/msal-net-token-cache-serialization?tabs=aspnetcore).
+- When the default internal token cache is used while using confidential client applications.
+  - The default token cache provided by MSAL is not designed to be performant when used in confidential client applications. Refer to [Token cache serialization in MSAL.NET](/entra/msal/dotnet/how-to/token-cache-serialization?tabs=aspnetcore) for more details.
 
 ## Metrics
 
 In addition to logging, MSAL exposes important metrics in <xref:Microsoft.Identity.Client.AuthenticationResult.AuthenticationResultMetadata?displayProperty=nameWithType>. See [Add monitoring around MSAL operations](high-availability.md#add-monitoring-around-msal-operations) for more details.
 
-`DurationTotalInMs` - total time spent in MSAL acquiring a token, including network calls and cache operations. Create an alert on overall high latency (more than 1 second). Note that the first ever token acquisition call usually makes an extra HTTP call.
+- `DurationTotalInMs` - total time spent in MSAL acquiring a token, including network calls and cache operations. Create an alert on overall high latency (more than 1 second). Note that the first ever token acquisition call usually makes an extra HTTP call.
+- `DurationInCacheInMs` - time spent loading or saving the token cache, which is customized by the app developer (for example, save to Redis). Create an alert on spikes.
```

Comment:
What is "spikes"? Do we have a link for what this is? Or a definition elsewhere we can refer to? Or are developers coming to this in full knowledge of what spikes are.